SignalServiceKit/src/Protos/Generated/SSKProto.swift

```diff
@@ -0,0 +1,3588 @@
+        }
+
+        @objc public func setContent(_ valueParam: Data) {
+            proto.content = valueParam
+        }
+
+        @objc public func build() throws -> SSKProtoEnvelope {
+            let wrapper = try SSKProtoEnvelope.parseProto(proto)
+            return wrapper
+        }
+    }
+
+    fileprivate let proto: SignalServiceProtos_Envelope
+
+    @objc public let type: SSKProtoEnvelopeType
+    @objc public let source: String
+    @objc public let sourceDevice: UInt32
+    @objc public let timestamp: UInt64
+
+    @objc public var relay: String? {
+        guard proto.hasRelay else {
+            return nil
+        }
+        return proto.relay
+    }
+    @objc public var hasRelay: Bool {
+        return proto.hasRelay
+    }
+
+    @objc public var legacyMessage: Data? {
+        guard proto.hasLegacyMessage else {
+            return nil
+        }
+        return proto.legacyMessage
+    }
+    @objc public var hasLegacyMessage: Bool {
+        return proto.hasLegacyMessage
+    }
+
+    @objc public var content: Data? {
+        guard proto.hasContent else {
+            return nil
+        }
+        return proto.content
+    }
+    @objc public var hasContent: Bool {
+        return proto.hasContent
+    }
+
+    private init(proto: SignalServiceProtos_Envelope,
+                 type: SSKProtoEnvelopeType,
+                 source: String,
+                 sourceDevice: UInt32,
+                 timestamp: UInt64) {
+        self.proto = proto
+        self.type = type
+        self.source = source
+        self.sourceDevice = sourceDevice
+        self.timestamp = timestamp
+    }
+
+    @objc
+    public func serializedData() throws -> Data {
+        return try self.proto.serializedData()
+    }
+
+    @objc public class func parseData(_ serializedData: Data) throws -> SSKProtoEnvelope {
+        let proto = try SignalServiceProtos_Envelope(serializedData: serializedData)
+        return try parseProto(proto)
+    }
+
+    fileprivate class func parseProto(_ proto: SignalServiceProtos_Envelope) throws -> SSKProtoEnvelope {
+        guard proto.hasType else {
+            throw SSKProtoError.invalidProtobuf(description: "\(logTag) missing required field: type")
+        }
+        let type = SSKProtoEnvelopeTypeWrap(proto.type)
+
+        guard proto.hasSource else {
+            throw SSKProtoError.invalidProtobuf(description: "\(logTag) missing required field: source")
+        }
+        let source = proto.source
+
+        guard proto.hasSourceDevice else {
+            throw SSKProtoError.invalidProtobuf(description: "\(logTag) missing required field: sourceDevice")
+        }
+        let sourceDevice = proto.sourceDevice
+
+        guard proto.hasTimestamp else {
+            throw SSKProtoError.invalidProtobuf(description: "\(logTag) missing required field: timestamp")
+        }
+        let timestamp = proto.timestamp
+
+        // MARK: - Begin Validation Logic for SSKProtoEnvelope -
+
```

Comment:
It seems like it would be really easy to blow away our customizations to this generated file. 

Could we separate any customizations into a separate file?

I guess since we're not actually using it anywhere at this point, I'll defer my bike-shedding on this for now.

Reply:
Yup, I'm considering removing support for customizations actually.  Let's see how it plays out.